cmds/wifi/wifi.go

```diff
@@ -38,34 +38,59 @@ func init() {
 	}
 }
 
+func generateConfig(a []string) (conf []byte, err error) {
+	// format of a: [essid, pass, id]
+	switch {
+	case len(a) == 3:
+		conf = []byte(fmt.Sprintf(eap, a[0], a[2], a[1]))
+	case len(a) == 2:
+		conf, err = passphrase.Run(a[0], a[1])
+		if err != nil {
+			return nil, fmt.Errorf("essid: %v, pass: %v : %v", a[0], a[1], err)
+		}
+	case len(a) == 1:
+		conf = []byte(fmt.Sprintf(nopassphrase, a[0]))
+	default:
+		flag.Usage()
+		os.Exit(1)
+	}
+	return
+}
+
 func main() {
+	go startServer()
+
+	// Service
 	var (
 		iface = flag.String("i", "wlan0", "interface to use")
-		essid string
 		conf  []byte
+		err   error
 	)
 
 	flag.Parse()
 	a := flag.Args()
 
-	switch {
-	case len(a) == 3:
-		essid = a[0]
-		conf = []byte(fmt.Sprintf(eap, essid, a[2], a[1]))
-	case len(a) == 2:
-		essid = a[0]
-		pass := a[1]
-		o, err := passphrase.Run(essid, pass)
-		if err != nil {
-			log.Fatalf("%v %v: %v", essid, pass, err)
+	if len(a) == 0 {
+		// Experimental Part
+		// if len(a) = 0, can use the web interface to get user's input
+		msg := <-ServerToServiceChan
+		a = append(a, msg.essid)
+		if msg.pass != "" {
+			a = append(a, msg.pass)
 		}
-		conf = o
-	case len(a) == 1:
-		essid = a[0]
-		conf = []byte(fmt.Sprintf(nopassphrase, essid))
-	default:
-		flag.Usage()
-		os.Exit(1)
+		if msg.id != "" {
+			a = append(a, msg.id)
+		}
```

Comment:
Question: why can't the msg just contain a slice of strings, then you don't have to do this append stuff.

Reply:
Fixed